src/Gui/Inventor/SoAutoZoomTranslation.h

```diff
@@ -23,7 +23,12 @@
 #ifndef GUI_SOAUTOZOOMTRANSLATION_H
 #define GUI_SOAUTOZOOMTRANSLATION_H
 
+#include <Gui/PreCompiled.h>
```

Comment:
Including a PreCompiled.h inside another header is an absolute no-go.

Reply:
OK.
I need to understand since these headers are already on precompiled.h, so why can't I use this header here? I thought precompiled is only used when compiling with PCH.

Reply:
The PreCompiled.h is only supposed to be included from .cpp files and never .h files because the file usually contains pre-processor defines that must not be visible outside the project PreCompiled.h is part of.

If a header of an extension module includes an application-specific header with an included PreCompiled.h it can hardly control which PreCompiled.h it actually uses. In most cases it includes the PreCompiled.h that is in the same directory as the application-header and thus sees any of the pre-processor defines which can cause undefined behaviour. If you are lucky you will run into compiler or linker errors and otherwise to hardly to find bugs.

Reply:
Thanks for explanation. I will change the PR accordingly.